GH-31: Port Dockerized integration tests to embedded integration tests

C0urante opened the pull request:
Addresses #31

The diff here is a little big but the logic is (hopefully) more straightforward:

- The Docker-based tests in `kcbq-connector/test` are completely removed
- A new integration test, `BigQuerySinkConnectorIT`, is added that replaces them by spinning up embedded instances of ZooKeeper, Kafka, Schema Registry, and Connect, then populates Kafka with the same data as before using the same serialization format (Schema Registry-based Avro), runs the connector with (nearly) the same configuration, and then verifies that the expected data made it to BigQuery
- The README is updated with instructions for running the tests. Go ahead, give it a try! They no longer accept properties files or command line arguments, but the existing environment variable-based approach is still supported

C0urante commented:
Ninth time's the charm--green build on Jenkins.

aakashnshah commented on kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/BaseConnectorIT.java:
```diff
@@ -149,16 +155,17 @@ protected void clearPriorTable(BigQuery bigQuery, String table) {
 
   protected void waitForCommittedRecords(
-      String connector, String topic, long numRecords, int numTasks
+      String connector, long numRecords, int numTasks, String topics
```
What was the idea behind this change?

C0urante replied:
Ah, good catch. I prepped this PR a while ago and some things got shuffled around between then and now; this change is unnecessary and I've reverted it in the latest push.

aakashnshah commented on kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/utils/BucketClearer.java:
```diff
@@ -0,0 +1,38 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BucketClearer {
```
Do you think it is useful to add a javadoc here?

C0urante replied:
Sure, added.

aakashnshah commented on kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/integration/utils/TableClearer.java:
```diff
@@ -0,0 +1,45 @@
+
+import static com.wepay.kafka.connect.bigquery.utils.TableNameUtils.table;
+
+public class TableClearer {
```
Here as well

C0urante replied:
👍

aakashnshah reviewed (commented):
Thanks Chris! This change is super useful and helpful for the future of this connector! It mostly looks good to me, just had a few simple questions.

C0urante commented:
Thanks @aakashnshah; I've addressed your points. This is ready for another round when you have time.

@criccomini @mtagle @whynick1 @stoynov96 would you guys like to take a look before this is merged?

aakashnshah reviewed (approved):
LGTM!
